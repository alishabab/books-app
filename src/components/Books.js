import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import getBooks from '../actions/book';

const Books = () => {
  const books = useSelector(state => state.books);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getBooks());
  }, [dispatch]);
  return (
    <ul>
      {books.map(book => <li key={book.id}>{book.volumeInfo.title}</li>)}
    </ul>
  );
};

export default Books;
