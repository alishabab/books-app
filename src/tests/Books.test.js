import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Books from '../components/Books';

const mockStore = configureStore([]);

describe('Books React-Redux Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      books: [
        {
          kind: 'books#volume',
          id: 'kIYSAAAAIAAJ',
          etag: 'AqoDY/Qlkdk',
          selfLink: 'https://www.googleapis.com/books/v1/volumes/kIYSAAAAIAAJ',
          volumeInfo: {
            title: 'Thesavrvs Lingvae Latinae Compendiarivs Or, A Compendious Dictionary of the Latin Tongue',
            subtitle: 'Designed Chiefly for the Use of the British Nations ...',
            authors: [
              'Robert Ainsworth',
            ],
            publishedDate: '1751',
            industryIdentifiers: [
              {
                type: 'OTHER',
                identifier: 'NYPL:33433075918759',
              },
            ],
            readingModes: {
              text: false,
              image: true,
            },
            printType: 'BOOK',
            categories: [
              'English language',
            ],
            maturityRating: 'NOT_MATURE',
            allowAnonLogging: false,
            contentVersion: '0.2.2.0.full.1',
            panelizationSummary: {
              containsEpubBubbles: false,
              containsImageBubbles: false,
            },
            imageLinks: {
              smallThumbnail: 'http://books.google.com/books/content?id=kIYSAAAAIAAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api',
              thumbnail: 'http://books.google.com/books/content?id=kIYSAAAAIAAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api',
            },
            language: 'en',
            previewLink: 'http://books.google.co.in/books?id=kIYSAAAAIAAJ&pg=PP433&dq=a&hl=&as_brr=7&cd=1&source=gbs_api',
            infoLink: 'https://play.google.com/store/books/details?id=kIYSAAAAIAAJ&source=gbs_api',
            canonicalVolumeLink: 'https://play.google.com/store/books/details?id=kIYSAAAAIAAJ',
          },
          saleInfo: {
            country: 'IN',
            saleability: 'FREE',
            isEbook: true,
            buyLink: 'https://play.google.com/store/books/details?id=kIYSAAAAIAAJ&rdid=book-kIYSAAAAIAAJ&rdot=1&source=gbs_api',
          },
          accessInfo: {
            country: 'IN',
            viewability: 'ALL_PAGES',
            embeddable: true,
            publicDomain: true,
            textToSpeechPermission: 'ALLOWED',
            epub: {
              isAvailable: false,
              downloadLink: 'http://books.google.co.in/books/download/Thesavrvs_Lingvae_Latinae_Compendiarivs.epub?id=kIYSAAAAIAAJ&hl=&output=epub&source=gbs_api',
            },
            pdf: {
              isAvailable: true,
              downloadLink: 'http://books.google.co.in/books/download/Thesavrvs_Lingvae_Latinae_Compendiarivs.pdf?id=kIYSAAAAIAAJ&hl=&output=pdf&sig=ACfU3U2mwu5GnWZ4gY5HXOBPPS5tn0uz3w&source=gbs_api',
            },
            webReaderLink: 'http://play.google.com/books/reader?id=kIYSAAAAIAAJ&hl=&as_brr=7&printsec=frontcover&source=gbs_api',
            accessViewStatus: 'FULL_PUBLIC_DOMAIN',
            quoteSharingAllowed: false,
          },
          searchInfo: {
            textSnippet: '<b>A</b> speig be [ bird ) Picus martius . membring ibeir former plunder and vi &amp; tory , <br> wified To spin out <b>a</b> discourse , Sermonem longiùe proA Jeel , Incantamentum , 2<br> . carmen magicum . for <b>a</b> civil war , &amp; quod plerique Sullani milites , ducěre .',
          },
        },
      ],
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <Router>
          <Books />
        </Router>
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should not dispatch action if books length is greather than 0', () => {
    expect(store.dispatch).toHaveBeenCalledTimes(0);
  });
});
