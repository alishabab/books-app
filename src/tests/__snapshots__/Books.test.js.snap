// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books React-Redux Component should render with given state from Redux store 1`] = `
<div>
  <div
    className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-8"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    >
      <a
        className="makeStyles-link-2"
        href="/books/kIYSAAAAIAAJ"
        onClick={[Function]}
      >
        <div
          className="MuiPaper-root MuiCard-root makeStyles-root-4 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <button
            className="MuiButtonBase-root MuiCardActionArea-root"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <img
              alt="Thesavrvs Lingvae Latinae Compendiarivs Or, A Compendious Dictionary of the Latin Tongue"
              className="MuiCardMedia-root makeStyles-image-5 MuiCardMedia-media MuiCardMedia-img"
              src="http://books.google.com/books/content?id=kIYSAAAAIAAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api"
              title="Contemplative Reptile"
            />
            <div
              className="MuiCardContent-root"
            >
              <h2
                className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                Thesavrvs Lingvae Latinae Compendiarivs Or, A Compendious Dictionary of the Latin Tongue
              </h2>
            </div>
            <span
              className="MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </a>
    </div>
  </div>
</div>
`;
